sensor: shell: Allow individual axis data to be processed

ubieda opened the pull request:
### Description
This PR allows the sensor-shell to process individual axis data by making two changes:
1. Allow _sensor_shell_ to request/process individual axis data.
2. Fix the default decoder to properly distinguish individual axis data from 3-axis data.

Fixes #72840

### Testing

- Build and Run
``` console
west build -b native_sim -t run samples/sensor/sensor_shell/ -- -DCONFIG_SAMPLES_SENSOR_SHELL_FAKE_SENSOR=y
```
- Console output
``` console
uart:~$ sensor get sensor@0 accel_x
channel type=0(accel_x) index=0 shift=0 num_samples=1 value=10660000000ns (0.000000)
uart:~$ sensor get sensor@0 accel_y
channel type=1(accel_y) index=0 shift=1 num_samples=1 value=11800000000ns (1.000000)
uart:~$ sensor get sensor@0 accel_z
channel type=2(accel_z) index=0 shift=2 num_samples=1 value=12800000000ns (2.000000)
uart:~$ sensor get sensor@0 accel_xyz
channel type=0(accel_x) index=0 shift=2 num_samples=1 value=15870000000ns (3.000000)
channel type=1(accel_y) index=0 shift=2 num_samples=1 value=15870000000ns (3.000000)
channel type=2(accel_z) index=0 shift=2 num_samples=1 value=15870000000ns (3.000000)
channel type=3(accel_xyz) index=0 shift=2 num_samples=1 value=15870000000ns, (3.000000, 3.000000, 3.000000)
uart:~$ sensor get sensor@0 gyro_x
channel type=4(gyro_x) index=0 shift=3 num_samples=1 value=25600000000ns (4.000000)
uart:~$ sensor get sensor@0 gyro_y
channel type=5(gyro_y) index=0 shift=3 num_samples=1 value=26640000000ns (5.000000)
uart:~$ sensor get sensor@0 gyro_z
channel type=6(gyro_z) index=0 shift=3 num_samples=1 value=27560000000ns (6.000000)
uart:~$ sensor get sensor@0 gyro_xyz
channel type=4(gyro_x) index=0 shift=3 num_samples=1 value=28880000000ns (7.000000)
channel type=5(gyro_y) index=0 shift=3 num_samples=1 value=28880000000ns (7.000000)
channel type=6(gyro_z) index=0 shift=3 num_samples=1 value=28880000000ns (7.000000)
channel type=7(gyro_xyz) index=0 shift=3 num_samples=1 value=28880000000ns, (7.000000, 7.000000, 7.000000)
uart:~$ 
```

ubieda committed:
sensor: shell: Allow individual axis data to be processed

For the supported channels, instead of just allowing 3-axis data being
printed out, allow individual channels to be processed (e.g: accel_x,
gyro_y, etc).

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
sensor: default_rtio_sensor: fix: Treat individual axis data as q31_t

Instead of assuming that an individual axis must contain all
data-values. Additionally, for determining that there's XYZ data, all
three channels must be present in the header.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda commented:
@ycsin could you confirm this fixes your issue? Thanks!

ycsin commented:
> @ycsin could you confirm this fixes your issue? Thanks!

Try to add a [test](https://github.com/zephyrproject-rtos/zephyr/pull/72972/commits/862833b38f1a6718e8dbb0d0f68ab50364b859d1)?

ubieda committed:
tests: sensor: generic_test: Remove special-handling for axis values

Since now they're treated as q31 values when individually queried, as
opposed to lump them in a triplet.

Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda committed:
samples: sensor:shell: test all channels

Loop through all of the channels and make sure that each
channel has an output.

Changes in patch originally present in #72972.

Signed-off-by: Yong Cong Sin <ycsin@meta.com>
Signed-off-by: Luis Ubieda <luisf@croxel.com>

ubieda commented:
> > @ycsin could you confirm this fixes your issue? Thanks!
> 
> 
> 
> Try to add a [test](https://github.com/zephyrproject-rtos/zephyr/pull/72972/commits/862833b38f1a6718e8dbb0d0f68ab50364b859d1)?

Done!

ycsin reviewed (approved):
Not expert in the sensor area, but LGTM since the test passed
